src/main/java/org/testng/internal/Parameters.java

```diff
@@ -264,35 +265,81 @@ else if (ITestResult.class.equals(c)) {
                                        MethodParameters params,
                                        XmlSuite xmlSuite) {
     List<Object> vResult = Lists.newArrayList();
-    for (int i = 0, j = 0; i < parameterTypes.length; i++) {
-      if (j < parameterNames.length) {
-        String p = parameterNames[j];
-        String value = params.xmlParameters.get(p);
-        if (null == value) {
-          // try SysEnv entries
-          value = System.getProperty(p);
+    if (isInjected) {
+      FilterOutInJectedTypesResult filterOutResult = FilterOutInJectedTypesFromOptionalValues(parameterTypes, optionalValues);
```

Comment:
It's a method, it should start with a lowercase: `filterOutInjectedTypesFromOptionalValues()`.

Reply:
oh, my nut head. will fix it.